samples/js/picasa/find-photos/src/main.js

```diff
@@ -0,0 +1,248 @@
+// @flow
+
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+import argv from 'yargs';
+import fit from 'aspect-fit';
+import {
+  DatasetSpec,
+  getTypeOfValue,
+  isSubtype,
+  makeListType,
+  makeStructType,
+  Map,
+  newStruct,
+  numberType,
+  PathSpec,
+  Set,
+  stringType,
+  Struct,
+  walk,
+} from '@attic/noms';
+
+const args = argv
+  .usage(
+    'Finds Noms Photo objects from output of picasa/slurp\n\n' +
+    'Usage: node . <in-path> <out-dataset>')
+  .demand(2)
+  .argv;
+
+main().catch(ex => {
+  console.error(ex);
+  process.exit(1);
+});
+
+const stringStruct = makeStructType('', {
+  'Q24t': stringType,
+});
+
+const sourceType = makeStructType('', {
+  'id': stringStruct,
+  'mediaQ24group': makeStructType('', {
+    'mediaQ24thumbnail': makeListType(makeStructType('', {
+      'height': numberType,
+      'width': numberType,
+      'url': stringType,
+    })),
+  }),
+  'gphotoQ24width': stringStruct,
+  'gphotoQ24height': stringStruct,
+  'published': stringStruct,
+  'updated': stringStruct,
+});
+
+const hasTitle = makeStructType('', {
+  'title': stringStruct,
+});
+
+const hasExifTime = makeStructType('', {
+  'gphotoQ24exiftime': makeListType(stringStruct),
+});
+
+const hasFaces = makeStructType('', {
+  'gphotoQ24shapes': makeStructType('', {
+    'gphotoQ24shape': makeListType(makeStructType('', {
+      'name': stringType,
+      'upperLeft': stringType,
+      'lowerRight': stringType,
+    })),
+  }),
+});
+
+const hasGeo = makeStructType('', {
+  'georssQ24where': makeStructType('', {
+    'gmlQ24Point': makeStructType('', {
+      'gmlQ24pos': stringStruct,
+    }),
+  }),
+});
+
+// From: https://developers.google.com/picasa-web/docs/2.0/reference#Parameters
+// More sizes are available (in fact, it appears you can just put any size you want in the URL),
+// but these suffice for us.
+const sizes = [320, 640, 1024, 1440, 1600];
+
+const clearLine = '\x1b[2K\r';
+
+async function main(): Promise<void> {
+  const inSpec = PathSpec.parse(args._[0]);
+  const pinnedSpec = await inSpec.pin();
+  if (!pinnedSpec) {
+    throw `Invalid input dataset: ${inSpec.path.dataset}`;
+  }
+  const [db, input] = await pinnedSpec.value();
+  if (!input) {
+    throw `Invalid input spec: ${inSpec.toString()}`;
+  }
+  const outSpec = DatasetSpec.parse(args._[1]);
+  const [outDB, output] = outSpec.dataset();
+  const result = [];
+
+  await walk(input, db, async (v: any) => {
+    if (v instanceof Struct && isSubtype(sourceType, getTypeOfValue(v))) {
+      const w = parseInt(v['gphotoQ24width']['Q24t'], 10);
```

Comment:
I decided not to because many of them are invalid identifiers and it looked more consistent to use same pattern everywhere in this file.